File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/utils/CoreUtils.java
package com.zumin.sudoku.common.core.utils;

import cn.hutool.core.util.ClassUtil;
import java.util.Set;
import javax.annotation.PostConstruct;
import lombok.RequiredArgsConstructor;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Component;

/**
 * 核心工具类
 */
@Component
@RequiredArgsConstructor
public class CoreUtils {

  public static String projectPackageName;

  private final Environment env;

  /**
   * 获取项目的所有Class
   *
   * @return Class集合
   */
  public static Set<Class<?>> getClasses() {
    return ClassUtil.scanPackage(projectPackageName);
  }

  @PostConstruct
  public void readConfig() {
    projectPackageName = env.getProperty("common.core.projectPackage", "com.zumin.sudoku");
  }
}

File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/utils/CoreUtils.kt
package com.zumin.sudoku.common.core.utils

import cn.hutool.core.util.ClassUtil
import com.zumin.sudoku.common.core.CommonCoreProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.stereotype.Component

/**
 * 核心工具类
 */
@Component
@EnableConfigurationProperties(CommonCoreProperties::class)
class CoreUtils(
  private val commonCoreProperties: CommonCoreProperties,
) {
  /**
   * 获取项目的所有Class
   *
   * @return Class集合
   */
  val projectClasses: Set<Class<*>>
    get() = ClassUtil.scanPackage(commonCoreProperties.projectPackage)
}